refactored format_event functions

ramonaspence opened the pull request:
Both format_event functions have repeating lines of code in an if/else block. This PR moves those repeated lines to outside of the if/else block, so that nothing is being repeated unnecessarily. This is discussed in more detail in #55. 

Edit:
I just added the change to online events so that their location is in the same format as listed on [Google](https://developers.google.com/search/docs/advanced/structured-data/event#online-event) for JSON+LD.

This provides search engines with more descriptive information about online events. It also avoids a lot of useless data for such events, like the address fields that are each None.

Closes #58

ramonaspence committed:
refactored fromat functions to be less repetitive

allella commented on app_functions.py:
```diff
@@ -54,100 +52,78 @@ def filter_events_by_tag(events, tags):
+                "tags": event.get('tags'), 
+                "startDate": event.get('time'), 
+                "url": event.get('url'), 
+                "identifier": f"urn:uuid:{event.get('uuid')}", 
```
@ramonaspence I see this hasn't changed, but what is this `f"urn` syntax doing / meaning?

allella replied:
I guess this wanted me to submit the review, even though @ commented to you. Did you get this now @ramonaspence  ?

ramonaspence replied:
@allella Yes! I see it now. 

The string with the f in front of it must have been left over from before we implemented truly unique id's. I'm looking for the issue or PR that we did that in to link it but I can't find it. 

In order to have that up to date, I just need to change it from `f"urn:uuid:{event.get('uuid')}"` to `event.get('uuid')`. It's currently grabbing the UUID in all_meetings.json, but adding `urn:uuid:` to the beginning of it.

ramonaspence committed:
change location of online event

ramonaspence commented:
I just added the change to online events so that their location is in the same format as listed on [Google](https://developers.google.com/search/docs/advanced/structured-data/event#online-event) for JSON+LD.

allella commented:
@ramonaspence I think the other PR created a merge conflict, but that should be easy to resolve.

Also, I had a general question about the syntax of line 123, which should appear in the unresolved notes on this issue.

allella commented:
@ramonaspence You can try to resolve the merge conflict through the GitHub interface.

Or, done a `git merge master` on this fix branch from the command line should merge the files and then you'll need to look for the <<<<<< blocks to figure out how to resolve them.  Once the code is resolved you can git add the resolved file and commit the resolution to this branch

ramonaspence committed:
edited function name in app.py to reflect the same change in app_functions.py

ramonaspence commented:
In this PR, I also changed the name of `format_ld_json` to `format_json_ld` to be more semantically correct, but I realized that I haven't made that change where it the function is being called in app.py. 

I'm going to go ahead and commit that change here since it's directly related to this PR.

ramonaspence committed:
Merge branch 'master' into fix/refactor_format_funcs

ramonaspence committed:
refactor format_json_ld so that online meetup events are recognized as online

the if/else logic should catch online meetup events as online as well as eventbrite events

ramonaspence commented:
@allella 
Alright, I've got the merge conflict resolved and I've double checked that everything to do with this PR has been completed. 

I was thinking that the uuid format you were asking about can be addressed in a separate issue and PR.

allella commented:
@ramonaspence this question would be a new issue and PR, but I do notice the _VirtualLocation > url_ values are pointing to the event homepage instead of a video stream.

I suspect not all events, like Code For Greenville, set those in advance, but I am curious if the event services provide those video stream URLs publicly in the APIs.

Again, this is beyond the scope of this issue to improve and differentiate a Place from a VirtualLocation, but that could be a next step. Or, if the streaming URL is only available some of the time, then perhaps we default to the event URL when there's no physical _Place_ and no stream URL.

Also, I haven't taken a deep dive in the Schema, but it's also possible the [VirtualLocation > url](https://schema.org/VirtualLocation) isn't meant to be a streaming URL, since it's possible there's some other property or entity type that better handles a stream.  In any case, this is all a good use case to learn more about Schema.org.

allella commented:
This PR looks good locally, so I'm going to merge and deploy it.

Thanks

ramonaspence commented:
@allella 

That's a great point to bring up. I'll keep reading more about Schema.org. I'll start more generally and then hone in on the VirtualLocation.
